walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,85 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration())
+      return new ArrayList<TimeRange>();
+    List<TimeRange> freeTimes = queryWithOptionalAttendees(events, request);
+    if (freeTimes.size() == 0 && request.getAttendees().size() > 0)
+      freeTimes = queryWithOnlyMandatoryAttendees(events, request);
```

Comment:
Omitting the braces on a single line if statement is handy sometimes, but accidental misuse has been the cause of many high-profile issues (e.g. https://www.imperialviolet.org/2014/02/22/applebug.html). We recommend always putting the curly braces.